stunnel/stunnel_cache.ml

```diff
@@ -187,10 +191,23 @@ let flush () =
       info "Flushed!")
 
 
-let connect ?use_fork_exec_helper ?write_to_log host port verify_cert =
+let rec connect ?use_fork_exec_helper ?write_to_log host port verify_cert =
   try
-    remove host port verify_cert
+    let s = remove host port verify_cert in 
+    if !Stunnel.legacy_protocol_and_ciphersuites_allowed
```

Comment:
This test should be in the `remove` function instead, since that can be (and indeed is) called directly from elsewhere.
I think this `Stunnel_cache.connect` function might well not be used at all at present.